crates/sui-indexer/src/sql_backfill.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::database::ConnectionPool;
+use diesel_async::RunQueryDsl;
+use futures::{stream, StreamExt};
+use std::time::Duration;
+use std::time::Instant;
+
+const CHUNK_SIZE: u64 = 10000;
+const MAX_THREADS: usize = 100;
+const SLEEP_BETWEEN_BATCHES_MS: u64 = 100;
+
+pub async fn run_sql_backfill(
+    sql: &str,
+    checkpoint_column_name: &str,
+    first_checkpoint: u64,
+    last_checkpoint: u64,
+    pool: ConnectionPool,
+) {
+    let cur_time = Instant::now();
+    let chunks: Vec<(u64, u64)> = (first_checkpoint..=last_checkpoint)
+        .step_by(CHUNK_SIZE as usize)
+        .map(|chunk_start| {
+            let chunk_end = std::cmp::min(chunk_start + CHUNK_SIZE - 1, last_checkpoint);
+            (chunk_start, chunk_end)
+        })
+        .collect();
+
+    stream::iter(chunks)
+        .for_each_concurrent(MAX_THREADS, |(start_id, end_id)| {
```

Comment:
`for_each_concurrent` doesn't leverage threads, but concurrent tasks so the const `MAX_THREADS` should probably be renamed